src/ratchet.rs

```diff
@@ -58,23 +60,23 @@ pub struct PreviousIterator {
 }
 
 impl Ratchet {
-    /// Creates a new ratchet with a randomly generated seed.
+    /// Creates a randomly-generated Skip Ratchet.
     ///
     /// # Examples
     ///
     /// ```
     /// use skip_ratchet::Ratchet;
     ///
-    /// let ratchet = Ratchet::new();
+    /// let ratchet = Ratchet::from_rng(&mut rand::thread_rng());
     /// ```
-    pub fn new() -> Self {
+    pub fn from_rng(rng: &mut impl CryptoRngCore) -> Self {
         // 32 bytes for the seed, plus two extra bytes to randomize small & medium starts
-        let seed = Hash::from_raw(rand::thread_rng().gen::<[u8; 32]>());
+        let seed = Hash::from_raw(rng.gen::<[u8; 32]>());
         let medium = Hash::from(&!seed);
         let small = Hash::from(&!medium);
 
-        let inc_med = rand::thread_rng().gen::<u8>();
-        let inc_small = rand::thread_rng().gen::<u8>();
+        let inc_med = rng.gen::<u8>();
```

Comment:
did `[gen()](https://docs.rs/rand/latest/rand/trait.Rng.html#method.gen)` guarantee that it fills the bytes if this is relied on ?

[fill_bytes()](https://rust-random.github.io/rand/rand_core/trait.RngCore.html#tymethod.fill_bytes) explicitly guarantees so and is defined by the [RngCore](https://rust-random.github.io/rand/rand_core/trait.RngCore.html).

Reply:
Ah, I see this:
> This method should guarantee that `dest` is entirely filled with new data, and may panic if this is impossible (e.g. reading past the end of a file that is being used as the source of randomness).
In the docs for `fill_bytes`.

Well, I assume that in most cases `gen` will just eventually call `fill_bytes`, but I'm just going to refactor into using `fill_bytes`. This way I can remove the `Rng` dependency, too.

Reply:
Yeah :+1: thats how we have it everywhere else. `Rng` is just some auto-impl and can move `rand` to dev-dependencies.